lib/db.c

```diff
@@ -1614,6 +1622,22 @@ grn_table_add(grn_ctx *ctx, grn_obj *table, const void *key, unsigned int key_si
           } GRN_TABLE_LOCK_END(ctx, table);
         });
         if (added) { *added = added_; }
+        if (id == GRN_ID_NIL) {
+          grn_obj buffer;
+          GRN_TEXT_INIT(&buffer, 0);
+          grn_inspect(ctx, &buffer, table);
```

Comment:
`grn_inspect()` includes key information for patricia trie. It's very large for large table. Did you try this with large table?